core/shared/src/main/scala/com/lynbrookrobotics/potassium/tasks/FiniteTask.scala

```diff
@@ -18,14 +16,15 @@ trait FiniteTaskFinishedListener {
   */
 abstract class FiniteTask extends Task { self =>
   private var running = false
-  private var listeners: List[WeakReference[FiniteTaskFinishedListener]] = List.empty
+  private var listener: Option[FiniteTaskFinishedListener] = None
 
   /**
     * Adds a task finish listener to listen to this task.
     * @param listener the listener of finish events
     */
-  def addFinishedListener(listener: FiniteTaskFinishedListener): Unit = {
-    listeners = WeakReference(listener) :: listeners
+  def setFinishedListener(newListener: FiniteTaskFinishedListener): Unit = {
+    assert(listener == None)
```

Comment:
Can we also print something incase the `-ea` flag is not set?

Reply:
The way Scala's assert is implemented, exceptions will be thrown no matter what the flag is set to.